engines/hdb/hdb.cpp

```diff
@@ -0,0 +1,1071 @@
+#include "common/config-manager.h"
+#include "common/random.h"
+
+#include "engines/util.h"
+
+#include "hdb/hdb.h"
+#include "hdb/ai.h"
+#include "hdb/file-manager.h"
+#include "hdb/gfx.h"
+#include "hdb/input.h"
+#include "hdb/lua-script.h"
+#include "hdb/map.h"
+#include "hdb/menu.h"
+#include "hdb/sound.h"
+#include "hdb/mpc.h"
+#include "hdb/window.h"
+
+#define CHEAT_PATCHES 0
+
+namespace HDB {
+
+HDBGame* g_hdb;
+
+HDBGame::HDBGame(OSystem *syst, const ADGameDescription *gameDesc) : Engine(syst), _gameDescription(gameDesc) {
+	g_hdb = this;
+	if (isPPC()) {
+		_screenWidth = 240;
+		_screenHeight = 320;
+		_screenDrawWidth = 240;
+		_screenDrawHeight = 320;
+		_progressY = 280;
+	} else {
+		_screenWidth = 640;
+		_screenHeight = 480;
+		_screenDrawWidth = _screenWidth - 160;
+		_screenDrawHeight = 480;
+		_progressY = _screenHeight - 64;
+	}
+
+	_format = Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0);
+	_systemInit = false;
+
+	_fileMan = nullptr;
+	_gfx = nullptr;
+	_lua = nullptr;
+	_menu = nullptr;
+	_map = nullptr;
+	_ai = nullptr;
+	_input = nullptr;
+	_sound = nullptr;
+	_window = nullptr;
+	_rnd = new Common::RandomSource("hdb");
+
+	_cheating = false;
+
+	_currentMapname[0] = _currentLuaName[0] = 0;
+	_lastMapname[0] = _lastLuaName[0] = 0;
+	_inMapName[0] = 0;
+
+	_timePlayed = _timeSlice = _prevTimeSlice = _timeSeconds = _tiempo = 0;
+
+	_currentOutSaveFile = NULL;
+	_currentInSaveFile = NULL;
+
+	_progressActive = false;
+
+	_monkeystone7 = STARS_MONKEYSTONE_7_FAKE;
+	_monkeystone14 = STARS_MONKEYSTONE_14_FAKE;
+	_monkeystone21 = STARS_MONKEYSTONE_21_FAKE;
+
+	syncSoundSettings();
+}
+
+HDBGame::~HDBGame() {
+	delete _fileMan;
+	delete _gfx;
+	delete _lua;
+	delete _menu;
+	delete _map;
+	delete _ai;
+	delete _input;
+	delete _sound;
+	delete _window;
+	delete _rnd;
+
+	delete _progressGfx;
+	_progressGfx = NULL;
+	delete _progressMarkGfx;
+	_progressMarkGfx = NULL;
+	delete _loadingScreenGfx;
+	_loadingScreenGfx = NULL;
+	if (_logoGfx) {
+		delete _logoGfx;
+		_logoGfx = NULL;
+	}
+	delete _debugLogo;
+	_debugLogo = NULL;
+}
+
+bool HDBGame::init() {
+	/*
+		Game Subsystem Initializations
+	*/
+
+	_systemInit = false;
+	_fileMan = new FileMan;
+	_gfx = new Gfx;
+	_lua = new LuaScript;
+	_menu = new Menu;
+	_map = new Map;
+	_ai = new AI;
+	_input = new Input;
+	_sound = new Sound;
+	_window = new Window;
+
+	// Init fileMan
+
+	if (!_fileMan->openMPC(getGameFile())) {
+		error("FileMan::openMPC: Cannot find the %s data file", getGameFile());
+	}
+	if (!_gfx->init()) {
+		error("Gfx::init: Couldn't initialize Gfx");
+	}
+	if (!_sound->init()) {
+		error("Window::init: Couldn't initialize Sound");
+	}
+	if (!_ai->init()) {
+		error("AI::init: Couldn't initialize AI");
+	}
+	if (!_window->init()) {
+		error("Window::init: Couldn't initialize Window");
+	}
+	if (!_input->init()) {
+		error("Input::init: Couldn't initialize Input");
+	}
+	if (!_lua->init()) {
+		error("LuaScript::init: Couldn't load the GLOBAL.LUA code.");
+	}
```

Comment:
It would be preferable to move all these init methods in their corresponding constructors

Reply:
We are following the original here. And such major refactorings will not bring anything, but potentially will introduce bugs